### PR TITLE
Update packages to rc3-23809

### PR DESCRIPTION
**Pull request by iamjasonp**

### CODE REVIEW

**Comment by iamjasonp** [comment]
@dotnet-bot test outerloop please

**Comment by iamjasonp** [comment]
@dotnet-bot test outerloop please

**Comment by iamjasonp** [comment]
@dotnet-bot test Linux Debug Build and Test

File: dir.props [hunk 1]
```diff
@@ -72,7 +72,7 @@
 
   <ItemGroup>
```
Comment (roncain) [inline dir.props]: Not necessary for this PR, but I learned our dir.props should not include the line:
```
  <DnuSourceList Include="https:%2F%2Fwww.myget.org/F/dotnet-coreclr/" />
```
This appears to be one of the reasons for build failure as we were moving to the new enterprise myget account (I think it wasn't forwarding properly when I tried to build). It is likely to cause us more problems in the future.  @iamjasonp -- do you think it makes sense to include that change in this PR?

**Comment by roncain** [comment]
I buddy-built, and though the OuterLoops passed, the packaging step failed for me with:
```
c:\OSS\wcf\Tools\Packaging.targets(651,5): error : Could not load runtime file: c:\OSS\wcf\packages/Microsoft.NETCore.Platforms/1.0.1-rc3-23729/runtime.json [c:\OSS\wcf\src\System.Private.ServiceMode
l\pkg\System.Private.ServiceModel.pkgproj]
```
It looks like Packaging.props contains that version reference.  Does it also need to be updated?  I'll clean and try again, but it could be due to the use of the myget reference to dotnet-coreclr.

**Comment by iamjasonp** [comment]
Yep - I'm seeing the error about System.Xml now, though oddly CI doesn't see it...

File: dir.props (hunk 1 above)
Comment (iamjasonp) [inline dir.props]: Let's include it.

**Comment by iamjasonp** [comment]
@dotnet-bot test outerloop please

File: dir.props [hunk 2]
```diff
@@ -85,34 +85,13 @@
...
-  </PropertyGroup>
-
   <!-- list of nuget package sources passed to dnu -->
   <ItemGroup Condition="'$(ExcludeInternetFeeds)' != 'true'">
     <!-- Need to escape double forward slash (%2F) or MSBuild will normalize to one slash on Unix. -->
 
-    <DnuSourceList Include="https:%2F%2Fdotnet.myget.org/F/dotnet-core/" />
-    <DnuSourceList Include="https:%2F%2Fdotnet.myget.org/F/dotnet-corefxtestdata/" />
-    <DnuSourceList Include="https:%2F%2Fdotnet.myget.org/F/dotnet-buildtools/" />
+    <DnuSourceList Include="https:%2F%2Fdotnet.myget.org/F/dotnet-core/api/v3/index.json" />
```
Comment (roncain) [inline dir.props]: Just to confirm -- these changes look like they are verbatim from CoreFx, right?

**Comment by roncain** [comment]
LGTM

File: dir.props (hunk 2 above)
Comment (iamjasonp) [inline dir.props]: Yep.

**Pull request merged**